tuna/rocmlir/rocmlir_worker.py

```diff
@@ -0,0 +1,165 @@
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+###############################################################################
+"""Builder class implements the worker interface. The purpose of this class is to run the
+tuningRunner.py command"""
+
+import sys
+import os
+from time import sleep
+import random
+import functools
+
+from sqlalchemy.inspection import inspect
+
+from tuna.dbBase.sql_alchemy import DbSession
+from tuna.worker_interface import WorkerInterface
+from tuna.rocmlir.rocmlir_tables import RocMLIRDBTables
+from tuna.utils.db_utility import session_retry, gen_insert_query
+
+
+class RocMLIRWorker(WorkerInterface):
+  """ The RocMLIR class implements the worker class. Its purpose is to run a command. It picks up
+  new jobs and when completed, sets the state to completed. """
+
+  def __init__(self, **kwargs):
+    """Constructor"""
+    self.dbt = None
+    super().__init__(**kwargs)
+    self.set_db_tables()
+    self.result_attr = [column.name for column in inspect(self.dbt.results).c]
+    self.result_attr.remove("insert_ts")
+    self.result_attr.remove("update_ts")
+
+  def set_db_tables(self):
+    """Initialize tables"""
+    self.dbt = RocMLIRDBTables(session_id=self.session_id)
+
+  def update_result_table(self, session, result_str):
+    """update results table with individual result entry"""
+    obj = self.dbt.results()
+
+    arch, config, perf_config, tflops = obj.parse(result_str)
+
+    print(f"arch = '{arch}', config = '{config}', perf_config = '{perf_config}', tflops = {tflops}",
+          file=sys.stderr)
+
+    obj.valid = 1
+    obj.session = self.dbt.session.id
+    obj.arch = arch
+    obj.config = self.job.config
+    obj.config_str = config
+    obj.perf_config = perf_config
+    obj.kernel_tflops = tflops
+
+    self.logger.info('Inserting results for job_id=%s', self.job.id)
+    query = gen_insert_query(obj, self.result_attr,
+                             self.dbt.results.__tablename__)
+    session.execute(query)
+    session.commit()
+    return True
+
+  def process_result(self, result_str: str):
```

Comment:
I don't know if I *need* the process_result/update_result_table combo, but that's what miopen did and it does work.  May be a simpler way.